econml/utilities.py

```diff
@@ -582,6 +583,25 @@ def get_input_columns(X, prefix="X"):
     return [f"{prefix}{i}" for i in range(len_X)]
 
 
+def get_feature_names_or_default(featurizer, feature_names):
+    if hasattr(featurizer, 'get_feature_names'):
+        # Get number of arguments, some sklearn featurizer don't accept feature_names
+        arg_no = len(inspect.getfullargspec(featurizer.get_feature_names).args)
+        if arg_no == 1:
+            return featurizer.get_feature_names()
+        elif arg_no == 2:
+            return featurizer.get_feature_names(feature_names)
+    # Featurizer doesn't have 'get_feature_names' or has atypical 'get_feature_names'
+    try:
+        # Get feature names using featurizer
+        dummy_X = np.empty((1, len(feature_names)))
```

Comment:
I think np.empty uses uninitialized memory so it might be better to use something like np.zeros in case the uninitialized memory happens to evaluate to NaN or infinity in some place and the transformer can't handle that.

Also it seems a bit weird that the type of the array that we're passing might not match what the transformer was trained on at all (e.g. if it was trained on strings and we pass numbers), but perhaps there's nothing better that we can do given that we're in an unusual situation anyway.

Reply:
I was hoping `np.empty` had random values so we don't run into issues at constant values. For example, `log(x)` has issues as 0, `1/x-1` has issues at 1. I guess the best is to pass in an array of `np.ones`, I will change it to that. For the future, maybe our estimators can record the featurized dimensions after fitting just like to record `d_x` now, but that required more complex changes. I will leave that for another PR.